src/cargo/util/toml/mod.rs

```diff
@@ -2231,8 +2231,8 @@ fn verify_lints(
     for (tool, lints) in lints {
         let supported = ["cargo", "clippy", "rust", "rustdoc"];
         if !supported.contains(&tool.as_str()) {
-            let supported = supported.join(", ");
-            anyhow::bail!("unsupported `{tool}` in `[lints]`, must be one of {supported}")
+            warnings.push(format!("unused manifest key `lints.{tool}`",));
```

Comment:
The request in the meeting was to specifically call out that this is reserved syntax and may break in the future.  From that, we should probably be more specific than unused key but unused lint tool info